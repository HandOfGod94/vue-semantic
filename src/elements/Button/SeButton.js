import Vue from 'vue'
import Component from 'vue-class-component'

@Component({
  name: 'se-button',
  props: {
    color: String,
    label: String,
    icon: String,
    basic: Boolean,
    inverted: Boolean,
    disabled: Boolean,
    loading: Boolean,
    size: String,
    float: String,
    compact: Boolean,
    fluid: Boolean,
    circular: Boolean
  }
})
class SeButton extends Vue {
  render (h) {
    return (
      <button class={this.classes}>
        { this.icon && <i class={`${this.icon} icon`}></i> }
        {this.label}
        {this.$slots.default}
      </button>
    )
  }

  get classes () {
    const classes = {
      [this.size]: this.size,
      compact: this.compact,
      circular: this.circular,

      ui: true,

      [`${this.float} floated`]: this.float,
      disabled: this.disabled,
      inverted: this.inverted,
      [this.color]: this.color,
      loading: this.loading,
      labeled: this.label,
      icon: this.icon,
      basic: this.basic,

      button: true
    }
    return classes
  }
}

export default SeButton
