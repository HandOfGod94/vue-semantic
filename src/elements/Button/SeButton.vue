<template>
  <component :is="as" :class="classObject">
    <i :class="`${this.icon} icon`" v-if="classObject.icon"></i>
    <slot name="label">
      {{label}}
    </slot>
    <slot></slot>
  </component>
</template>

<script>
export default {
  name: 'se-button',
  props: {
    as: { type: String, default: 'button' },
    active: Boolean,
    attach: String,
    color: String,
    label: String,
    icon: String,
    basic: Boolean,
    inverted: Boolean,
    disabled: Boolean,
    loading: Boolean,
    size: String,
    float: String,
    compact: Boolean,
    fluid: Boolean,
    circular: Boolean
  },
  computed: {
    classObject () {
      return {
        [this.size]: this.size,
        compact: this.compact,
        circular: this.circular,
        ui: true, // Semantic ui class
        [`${this.attach} attached`]: this.attach,
        [`${this.float} floated`]: this.float,
        disabled: this.disabled,
        inverted: this.inverted,
        [this.color]: this.color,
        loading: this.loading,
        labeled: this.label,
        icon: this.icon,
        basic: this.basic,
        button: true
      }
    }
  }
}
</script>
