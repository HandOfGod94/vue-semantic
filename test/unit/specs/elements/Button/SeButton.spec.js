import SeButton from '@/elements/Button/SeButton'
import { mount } from '@vue/test-utils'

describe('SeButton snapshot test', () => {
  it('should match with exisitng snapshot', () => {
    const wrapper = mount(SeButton)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match with button having icon and label snapshot', () => {
    const data = { propsData: { icon: 'mail', label: 'Send Mail' } }
    const wrapper = mount(SeButton, data)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('SeButton classes test', () => {
  it('should have correct classes for primary color', () => {
    const data = { propsData: { color: 'primary' } }
    const wrapper = mount(SeButton, data)
    expect(wrapper.classes()).toContain('primary')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have correct classes for custom color', () => {
    const data = { propsData: { color: 'orange' } }
    const wrapper = mount(SeButton, data)
    expect(wrapper.classes()).toContain('orange')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have correct classes for circular variation', () => {
    const data = { propsData: { circular: true } }
    const wrapper = mount(SeButton, data)
    expect(wrapper.classes()).toContain('circular')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have correct classes for basic variation', () => {
    const data = { propsData: { basic: true } }
    const wrapper = mount(SeButton, data)
    expect(wrapper.classes()).toContain('basic')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have "icon" class', () => {
    const data = { propsData: { icon: 'play' } }
    const wrapper = mount(SeButton, data)
    expect(wrapper.contains('i')).toBe(true)
    expect(wrapper.classes()).toContain('icon')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should have correct classes for float', () => {
    const data = { propsData: { float: 'right' } }
    const wrapper = mount(SeButton, data)
    expect(wrapper.classes()).toContain('right')
    expect(wrapper.classes()).toContain('floated')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
