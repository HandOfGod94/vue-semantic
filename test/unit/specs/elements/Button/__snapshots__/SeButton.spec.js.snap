// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeButton classes test should have "icon" class 1`] = `
<button class="ui icon button">
    <i class="play icon"></i>
</button>
`;

exports[`SeButton classes test should have correct classes for basic variation 1`] = `<button class="ui basic button"></button>`;

exports[`SeButton classes test should have correct classes for circular variation 1`] = `<button class="circular ui button"></button>`;

exports[`SeButton classes test should have correct classes for custom color 1`] = `<button class="ui orange button"></button>`;

exports[`SeButton classes test should have correct classes for float 1`] = `<button class="ui right floated button"></button>`;

exports[`SeButton classes test should have correct classes for primary color 1`] = `<button class="ui primary button"></button>`;

exports[`SeButton snapshot test should match with button having icon and label snapshot 1`] = `
<button class="ui labeled icon button">
    <i class="mail icon"></i>Send Mail</button>
`;

exports[`SeButton snapshot test should match with exisitng snapshot 1`] = `<button class="ui button"></button>`;
